prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -233,32 +344,123 @@ static void handle_connreq(struct poll_fd_mgr *poll_mgr,
 		return;
 	}
 
+	ret = ofi_recv_socket(sock, &conn_req, sizeof(conn_req), MSG_WAITALL);
+	if (ret) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL, "IO failed\n");
+		goto err1;
+	}
+
+	if (conn_req.version != OFI_CTRL_VERSION) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL, "protocol version mismatch\n");
+		goto err1;
+	}
+
+	if (conn_req.type != ofi_ctrl_connreq) {
+		FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL, "invalid message received\n");
+		goto err1;
+	}
+
+	cm_data_sz = ntohs(conn_req.seg_size);
+	if (cm_data_sz) {
+		if (cm_data_sz > TCPX_MAX_CM_DATA_SIZE)
+			goto err1;
+
+		ret = ofi_recv_socket(sock, poll_info->cm_data, cm_data_sz, MSG_WAITALL);
+		if (ret) {
+			FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL, "IO failed for cm-data\n");
+			goto err1;
+		}
+	}
```

Comment:
Can the above code be moved to a common location and merged with proc_recv_resp()?  They look very similar.